lib/fdo.c

```diff
@@ -843,151 +854,87 @@ static bool _STATE_DI(void)
 
 	fdo_prot_di_init(&g_fdo_data->prot, g_fdo_data->devcred);
 
-	fdo_ip_address_t *manIPAddr = NULL;
-
-#if defined(TARGET_OS_LINUX) || defined(TARGET_OS_MBEDOS) ||                   \
-    defined(TARGET_OS_OPTEE)
-	char *mfg_dns = NULL;
-	char *buffer = NULL;
-	bool is_mfg_addr = false;
-
-	fsize = fdo_blob_size((char *)MANUFACTURER_IP, FDO_SDK_RAW_DATA);
+	fsize = fdo_blob_size((char *)MANUFACTURER_ADDR, FDO_SDK_RAW_DATA);
 	if (fsize > 0) {
 		buffer = fdo_alloc(fsize + 1);
 		if (buffer == NULL) {
 			LOG(LOG_ERROR, "malloc failed\n");
 			goto end;
 		}
 
-		if (fdo_blob_read((char *)MANUFACTURER_IP, FDO_SDK_RAW_DATA,
+		if (fdo_blob_read((char *)MANUFACTURER_ADDR, FDO_SDK_RAW_DATA,
 				  (uint8_t *)buffer, fsize) == -1) {
-			LOG(LOG_ERROR, "Failed to read Manufacture DN\n");
-			fdo_free(buffer);
+			LOG(LOG_ERROR, "Failed to read Manufacture address\n");
 			goto end;
 		}
 
 		buffer[fsize] = '\0';
-		manIPAddr = fdo_ipaddress_alloc();
 
-		if (!manIPAddr) {
-			LOG(LOG_ERROR, "Failed to alloc memory\n");
-			ERROR()
-			fdo_free(buffer);
+		// the expected format is '{http/https}://{IP/DNS}:port', port is optional
+		result = sscanf(buffer, "%5[^:]://%99[^:]:%d/",
```

Comment:
It would be better to create a wrapper function to read the respective  mfg_prot / mfg_ip / mfg_dns / mfg_port from given buffer. This way, you would be able to write unit-test for that function.